cmd/podman/images/scan.go

```diff
@@ -0,0 +1,156 @@
+package images
+
+import (
+	"context"
+	"fmt"
+	"github.com/containers/podman/v2/cmd/podman/common"
+	"github.com/containers/podman/v2/cmd/podman/registry"
+	"github.com/containers/podman/v2/pkg/domain/entities"
+	"github.com/containers/podman/v2/pkg/specgen"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"os"
+	"strings"
+)
+
+var (
+	scanDescription = `Scan a container image for known vulnerabilities.  The image name or digest can be used.`
+	scanCommand     = &cobra.Command{
+		Use:     "scan [run-options] IMAGE [scanner-options]",
+		Short:   "Scan a container image for known vulnerabilities",
+		Long:    scanDescription,
+		RunE:    scan,
+		Example: `podman scan centos:latest`,
+	}
+
+	imageScanCommand     = &cobra.Command{
+		Use:     "scan [run-options] IMAGE [scanner-options]",
+		Short:   "Scan a container image for known vulnerabilities",
+		Long:    scanDescription,
+		RunE:    scan,
+		Example: `podman image scan centos:latest`,
+	}
+)
+
+var (
+	scanOptions = entities.ImageScanOptions{
+		ContainerRunOptions: entities.ContainerRunOptions{
```

Comment:
I don't think we should give full access to `ContainerRunOptions`.  I don't want `image scan` to be a way of running containers but to scan a specific image.  The fact that we're running a container should be hidden from the user as much as possible so that they can focus on the security scans and not worry about the (gigantic) amount of options available for running a container.

Let's start with a minimal use case.  If we need more functionality, we can add it in the future.

Reply:
This feedback helps a lot, thanks! I was waffling between a minimum profile (as you're suggesting) and a opening up the run options (as implemented now)... I'll make the adjustments :+1: